### PR TITLE
Don't treat empty output filename specially (fix #27)

### LINKED ISSUE #27: Why treat empty filenames specially?
opustags complains if a filename argument is empty. Most programs don't do this, they will just give an error when they try to open a file with an empty name.

### PR DESCRIPTION
**Pull request by rrthomas**

### CODE REVIEW

**Review by fmang — CHANGES_REQUESTED** [review]
You’re breaking this code in ot::run:

```c++
	/* Read-only mode. */
	if (opt.path_out.empty())
		return process(reader, nullptr, opt);
```

If someone specified `-o ""` then this code will just treat it as if -o wasn’t specified at all.

I’ll upgrade the build system to use C++17, so you’ll be able to use `std::optional` instead of keeping a boolean.

**Comment by rrthomas** [comment]
I've already pushed a PR that rewrites the code that was broken; but I agree that an `optional` would be better.

**Comment by fmang** [comment]
You should now be able to use std::optional: 5ea2db2d6d57315c99922684a67005fbcc62fc3d

**Comment by rrthomas** [comment]
I've updated the branch to use `std::optional`.

**Comment by fmang** [comment]
Pretty cool! Why bother with all the casts though? If I’m not mistaken:

1. x = std::optional(y) could be written x = y.
2. x.value() == y could be written x == y.
3. x.value().c_str() could be written x->c_str().

**Comment by rrthomas** [comment]
Because I don't know much C++! PR updated.

File: t/cli.cc [hunk 1]
```diff
@@ -115,6 +114,8 @@ void check_bad_arguments()
 	error_case({"opustags", "-o", "x", "--output", "y", "z"},
 	           "Cannot specify --output more than once.", "double output");
 	error_code_case({"opustags", "-S", "x"}, "Malformed tag: INVALID", ot::st::error, "attempt to read invalid argument with -S");
+	error_case({"opustags", "-o", "", "--output", "y", "z"},
+	           "Cannot specify --output more than once.", "double output with first filename empty");
```
Comment (fmang) [inline t/cli.cc:118]: Very nice edge case!

**Comment by fmang** [comment]
Thanks!

**Pull request merged**